src/Proxy/ProxyGenerator.php

```diff
@@ -940,7 +941,12 @@ private function generateMethods(ClassMetadata $class)
             if ($this->isShortIdentifierGetter($method, $class)) {
                 $identifier = lcfirst(substr($name, 3));
                 $fieldType  = $class->getTypeOfField($identifier);
-                $cast       = in_array($fieldType, ['integer', 'smallint'], true) ? '(int) ' : '';
+
+                if (ClassUtils::containsEnumType($method->getReturnType())) {
```

Comment:
I'm afraid this is only half-baked solution as without method's return type we still do not know what's inside. Maybe it'd be more efficient to generate an `instanceof \BackedEnum` check for the `return` statement?

Reply:
Check instance in template, like this ?
from:
```   
        if ($this->__isInitialized__ === false) {
            return (int)  parent::getId();
        }
```

to:
```   
        if ($this->__isInitialized__ === false) {
            return  parent::getId() instanceof \BackedEnum ? (int) parent::getId() :  parent::getId();
        }
```